datalad/support/annexrepo.py

```diff
@@ -0,0 +1,67 @@
+"""
+Interface to git-annex by Joey Hess.
+See https://git-annex.branchable.com/.
+"""
+__author__ = 'Benjamin Poldrack'
+
+import os
+from os.path import join, exists
+
+from gitrepo import GitRepo
+import datalad.log
+
+from datalad.cmd import Runner
+
+
+class AnnexRepo(GitRepo):
+    """
+    Representation of an git-annex repository.
+
+    """
+
+    def __init__(self, path, url=None):
+        """
+        AnnexRepo is initialized by giving a path to the annex.
+        If no annex exists at that location, a new one is created.
+        Optionally give url to clone from.
+
+        :param path: path to git-annex repository
+        :param url: valid git url. See http://www.kernel.org/pub/software/scm/git/docs/git-clone.html#URLS
+        :return:
+        """
+        super(AnnexRepo, self).__init__(path, url)
+
+        self.cmd_call_wrapper = Runner()
+        # TODO: Concept of when to set to "dry". Includes: What to do in gitrepo class?
+
+
+        # Check whether an annex already exists at destination
+        if not exists(join(self.path, '.git', 'annex')):
+            datalad.log.lgr.debug('No annex found in %s. Creating a new one ...' % self.path)
+            self.annex_init()
+
+    def annex_init(self):
+        """
+        Invokes command 'git annex init'.
+
+        :return:
+        """
+
+        oldCWD = os.getcwd()  # save to restore afterwards
+        os.chdir(self.path)
+
+
+        status, output = self.cmd_call_wrapper.getstatusoutput('git annex init')
+        datalad.log.lgr.info('\"git annex init\" outputs:\n %s' % output)
```

Comment:
I wasn'T getting getting anything on stdout from getstatusoutput. I just
want to see what it says by now. Later on, we should redirect and catch the
output of git/git-annex to
1. parse and handle it and
2. Provide an output to the users that doesn't require them to know about
git/git-annex


So, I will fix this, as soon as it is clear where to put the output.

2015-02-23 17:10 GMT+01:00 Yaroslav Halchenko <notifications@github.com>:

> In datalad/support/annexrepo.py
> <https://github.com/datalad/datalad/pull/42#discussion_r25175843>:
>
> > +            datalad.log.lgr.debug('No annex found in %s. Creating a new one ...' % self.path)
> > +            self.annex_init()
> > +
> > +    def annex_init(self):
> > +        """
> > +        Invokes command 'git annex init'.
> > +
> > +        :return:
> > +        """
> > +
> > +        oldCWD = os.getcwd()  # save to restore afterwards
> > +        os.chdir(self.path)
> > +
> > +
> > +        status, output = self.cmd_call_wrapper.getstatusoutput('git annex init')
> > +        datalad.log.lgr.info('\"git annex init\" outputs:\n %s' % output)
>
>
>    1. just define lgr = logging.getLogger("datalad.annexrepo") on top
>    globally
>    2. no need to escape " within ' string.
>     3. Actually no need to log here anything anyways since that
>    getstatusoutput does that anyways
>
> I might later comment on re-using the same Runner instance here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/datalad/datalad/pull/42/files#r25175843>.
>